Automattic/jetpack
# ContentLens: replace AI Post Excerpt with core post excerpt

This PR registers the new AI Excerpt post panel and removes the core one since the API currently doesn't allow extending it.
The whole implementation is hidden behind the `ai-content-lens` feature.

Fixes https://github.com/Automattic/jetpack/issues/32843

## Proposed changes:

* Content Lens: replace AI Post Excerpt with core post excerpt

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Open the post sidebar
* Confirm the Excerpt panel is there when:
  * `beta` extensions are enabled
  * `AI Assistant block` is enabled

Settings | Enable/Disable Ai Assistant block
--------|-----------
<img width="513" alt="Screenshot 2023-09-05 at 16 41 05" src="https://github.com/Automattic/jetpack/assets/77539/d4ad0046-703e-44c9-b70e-6eae010fbf11"> | <img width="785" alt="Screenshot 2023-09-05 at 16 41 24" src="https://github.com/Automattic/jetpack/assets/77539/c7825239-5b9d-476a-b3e3-10b249aadc44">

You can differentiate the Excerpt implementation because the Jetpack one adds the AI Icon there

core Excerpt | Jetpack Excerpt
----------|----------
<img width="328" alt="Screenshot 2023-09-05 at 16 38 43" src="https://github.com/Automattic/jetpack/assets/77539/6cc381d5-9c67-40c6-ad37-d54510ed7637"> | <img width="336" alt="Screenshot 2023-09-05 at 16 36 59" src="https://github.com/Automattic/jetpack/assets/77539/4820c763-cf9f-4013-bbb5-0017138120bb">


* Confirm the Jetpack AI Excerpt works as usual 
  * Set a custom excerpt
  * Save the post 
  * Go to a front-page page that shows the post excerpt. Usually blog posts page.
